### PR TITLE
Docs: Add building and testing to contributing guide

### PR DESCRIPTION
**Pull request by twpayne**
This should help new contributors get started a little quicker.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,6 +6,20 @@ Contributions to this project are [released](https://help.github.com/articles/gi
...
+
+2. Ensure that `<extraction-root>/codeql` is in your `PATH`.
+
+3. Clone this repository into `<extraction-root>/codeql-go` and change to this directory.
+
+4. To build, run `make`.
+
+5. To run the full test suite, run `make test`. This will take approximately one hour.
+
+6. To run an individual test, run `codeql test run <directory>`, for example `codeql test run ql/test/query-tests/Security`.
```
Comment (sauyon) [inline CONTRIBUTING.md:21]: You can also run individual queries by passing the query file when there are multiple queries in a directory.

**Review by sauyon — COMMENTED** [review]
Thanks, this will definitely be helpful for the future.

One comment, but otherwise looks great.

File: CONTRIBUTING.md (hunk 1 above)
Comment (twpayne) [reply CONTRIBUTING.md]: Thanks, done.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,6 +6,22 @@ Contributions to this project are [released](https://help.github.com/articles/gi
...
+
+3. Clone this repository into `<extraction-root>/codeql-go` and change to this directory.
+
+4. To build, run `make`.
+
+5. To run the full test suite, run `make test`. This will take approximately one hour.
+
+6. To run all tests in a directory and its subdirectories, run `codeql test run <directory>`, for example `codeql test run ql/test/query-tests/Security`.
+
+7. To run an individual test, run `codeql test run <filename>`, where `<filename>` is a `.qlref` file, for example `codeql test run ql/test/query-tests/Security/CWE-020/MissingRegexpAnchor/MissingRegexpAnchor.qlref`.
```
Comment (sauyon) [inline CONTRIBUTING.md:23]: Tests can also be `.ql` files, so this needs a tiny modification:

```suggestion
7. To run an individual test, run `codeql test run <filename>`, where `<filename>` is a `.ql` or `.qlref` file, for example `codeql test run ql/test/query-tests/Security/CWE-020/MissingRegexpAnchor/MissingRegexpAnchor.qlref`.
```

Comment (twpayne) [reply CONTRIBUTING.md]: Thanks, done.

**Review by sauyon — APPROVED** [review]
:+1: thanks again!

**Pull request merged**